transformations/transformations/data_offload.py

```diff
@@ -429,14 +428,9 @@ def process_driver(self, routine, successors):
             new_imports += as_tuple(Import(k, symbols=tuple(Variable(name=s, scope=routine) for s in v)))
 
         # add new imports to driver subroutine sepcification
-        import_pos = 0
-        if (old_imports := FindNodes(Import).visit(routine.spec)):
-            import_pos = routine.spec.body.index(old_imports[-1]) + 1
-        if new_imports:
-            routine.spec.insert(import_pos, Comment(text=
+        routine.spec.prepend(new_imports)
+        routine.spec.prepend(Comment(text=
                '![Loki::GlobalVarOffload].....Adding global variables to driver symbol table for offload instructions'))
-            import_pos += 1
-            routine.spec.insert(import_pos, new_imports)
```

Comment:
This change modifies the way the module variable imports are added to the driver routine. The original inserts them to the specification part. The problem with this is that it may add the imports after `IMPLICIT NONE`, which results in errors when compiling the transformed code. The suggested code avoids this by prepending the imports.